test.cpp

```
@@ -0,0 +1,135 @@
+    }
+}
+
+float calculateAverage(Student* students, int count) {
+    float sum;
+    for (int i = 0; i < count; i++) {
+        sum += students[i].grade; // 錯誤：sum 未初始化
+    }
+    return sum / count; // 邏輯錯誤：若 count = 0，會除以 0
+}
+
+void searchStudent(Student* students, int count, string name) {
```

Cppcheck (2.10-2)

[performance] Function parameter 'name' should be passed by const reference.

(passedByValue)